### PR TITLE
chore: update repo documentation

### PR DESCRIPTION
**Pull request by mohnoor94**
# Situation
The SDK Toolkit README was outdated and lacked comprehensive information about its purpose, components, and usage for developers.

# Task
Update the README to provide clear, concise documentation of the SDK Toolkit, including its components, usage, and contribution guidelines.

# Action
- Restructured README with clear sections
- Added comprehensive toolkit components list
- Included SDKs table with Maven and documentation links
- Expanded "Getting Started" with prerequisites and build instructions

# Testing
N/A

# Results
- Improved documentation clarity
- Easier developer onboarding
- Clear toolkit capabilities overview

# Notes
**Internal Ticket**: SDK-1109

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,124 @@
-# Expedia Group<sup>TM</sup> SDK for Java
+# Expedia Group Java SDK Foundations
 
-The Expedia Group SDK for Java allows Expedia Group partners to easily build Java applications that leverage the Expedia Group<sup>TM</sup> platform.
+The **Expedia Group Java SDK Foundations** provides the core components and tools for building and maintaining our SDK libraries.
+These foundations enable developers to create consistent, high-quality Java SDKs that interact with the Expedia Group<sup>TM</sup> platform.
```
Comment (anssari1) [inline README.md]: Aren't we dropping TM superscripts?

Comment (mohnoor94) [reply README.md]: Removed

**Review by anssari1 — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -1,11 +1,124 @@
...
+---
+
+## Getting Started
+
+### Prerequisites
+
+To generate a new SDK, ensure you have the following:
+
+1. **OpenAPI Specification** for the API you wish to generate an SDK for.
+2. Use the [Spec Transformer](https://github.com/ExpediaGroup/spec-transformer) if transformation is required. Learn more about
```
Comment (dtayeh) [inline README.md:40]: how do I know if the transformation is required? I looked at the transformer readme file, and it doesn't describe what is the kind of transformation it is doing either.

File: README.md [hunk 3]
```diff
@@ -1,11 +1,124 @@
...
+
+Follow these steps to generate an SDK:
+
+1. Clone this repository:
+   ```bash
+   git clone https://github.com/ExpediaGroup/expediagroup-java-sdk.git
+   cd expediagroup-java-sdk
+   ```
+
+2. Build the SDK Generator:
```
Comment (dtayeh) [inline README.md:55]: Isn't this missing the next step to actually generate an SDK? The title say generating SDKs, but the steps end at building the generator

File: README.md [hunk 4]
```diff
@@ -1,11 +1,124 @@
...
+
+```groovy
+implementation 'com.expediagroup:your-sdk-name:x.y.z'
+```
+
+Replace `your-sdk-name` and `x.y.z` with the appropriate SDK name and version.
+
+--- 
+
+This addition ensures developers using Gradle have clear instructions for integrating the SDK.
```
Comment (dtayeh) [inline README.md]: I'm a bit confused by this line, what addition?

File: README.md (hunk 2 above)
Comment (mohnoor94) [reply README.md:40]: There is another ticket to enhance the documentation of Spec Transformer.

File: README.md (hunk 3 above)
Comment (mohnoor94) [reply README.md:55]: Step 3 refers to the SDK-generating steps inside the generator package.

File: README.md (hunk 4 above)
Comment (mohnoor94) [reply README.md]: removed.

File: README.md (hunk 3 above)
Comment (dtayeh) [reply README.md:55]: But still it makes more sense to mention the basic command here:
```
cd openapi
../../mvnw exec:java
```

Then reference the generator package for more details, right now step 3 gives the impression that this is something additional, and this section should suffice to generate an SDK on its own without following the link.

Comment (mohnoor94) [reply README.md:55]: Right. Updated.

**Review by dtayeh — DISMISSED** [review]

File: README.md [hunk 5]
```diff
@@ -1,11 +1,135 @@
...
+|-------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------|
+| [Rapid SDK](https://github.com/ExpediaGroup/rapid-java-sdk) | [![Maven Central Version](https://img.shields.io/maven-central/v/com.expediagroup/rapid-sdk?style=for-the-badge&logo=expedia&logoColor=%23fddb32&label=Rapid%20SDK&labelColor=%23181e3b&color=%23fddb32)](https://central.sonatype.com/artifact/com.expediagroup/rapid-sdk) | [Rapid SDK on Developer Hub](https://developers.expediagroup.com/docs/products/rapid/sdk/java) |
```
Comment (anssari1) [inline README.md:14]: I will add XAP SDK after we publish it today..

**Review by anssari1 — APPROVED** [review]

**Review by dtayeh — APPROVED** [review]

**Pull request merged**